wasp/src/main/java/com/orhanobut/wasp/VolleyNetworkStack.java

```diff
@@ -159,22 +158,34 @@ public void onErrorResponse(VolleyError error) {
     }
   }
 
-  private static class VolleyRequest<T> extends Request<T> {
+  private static class VolleyRequest extends Request<Response> {
 
     /**
      * Charset for request.
      */
     private static final String PROTOCOL_CHARSET = "UTF-8";
 
-    private final VolleyListener<T> listener;
+    private final VolleyListener<Response> listener;
+    private final RequestFuture<Object> future;
     private final String requestBody;
     private final String url;
     private final Type responseObjectType;
     private final RequestCreator waspRequest;
 
-    public VolleyRequest(int method, String url, RequestCreator request, VolleyListener<T> listener) {
```

Comment:
I think we should refactor this class, having 2 constructor which almost does the same job could be hard to maintain, I will suggest, let's change the constructor signature and adjust it to volley requirements. Request class only requires ErrorListener, Both VolleyListener and RequestuFuture already implement ErrorListener, passing them would be fine. so let's meet this requirement and handle response in the anonymous class, since sync and async handles it differently. 

VolleyRequest, we can remove the listener and requestfuture fields, no longer required, passing errorListener would be fine
```java

    //REMOVED private final VolleyListener<Response> listener;  
    //REMOVED private final RequestFuture<Object> future;

    public VolleyRequest(int method, String url, RequestCreator request,
                         com.android.volley.Response.ErrorListener errorListener) {
      super(method, url, errorListener);
      this.url = url;
      this.requestBody = request.getBody();
      this.responseObjectType = request.getMethodInfo().getResponseObjectType();
      this.waspRequest = request;
    }
```
request call and response handling for async
```java
    final VolleyListener<T> listener = new VolleyListener<>(waspCallback, url);
    Request<Response> request = new VolleyRequest<Response>(method, url, waspRequest, listener){
      @Override
      protected void deliverResponse(Response response) {
        super.deliverResponse(response);
        listener.onResponse(response);
      }
    };
```
for sync call
```java
    final RequestFuture<T> future = RequestFuture.newFuture();
    Request<Response> request = new VolleyRequest<Response>(
        getMethod(requestCreator.getMethod()),
        requestCreator.getUrl(),
        requestCreator,
        future
    ){
      @Override
      protected void deliverResponse(Response response) {
        super.deliverResponse(response);
        future.onResponse(response.getResponseObject());
      }
    };
```